packages/react-query-next-experimental/src/ReactQueryStreamedHydration.tsx

```diff
@@ -0,0 +1,87 @@
+'use client'
+
+import type { DehydratedState, QueryClient } from '@tanstack/react-query'
+import { dehydrate, hydrate, useQueryClient } from '@tanstack/react-query'
+import * as React from 'react'
+import type { HydrationStreamProviderProps } from './HydrationStreamProvider'
+import { createHydrationStreamProvider } from './HydrationStreamProvider'
+
+const stream = createHydrationStreamProvider<DehydratedState>()
+
+/**
+ * This component is responsible for:
+ * - hydrating the query client on the server
+ * - dehydrating the query client on the server
+ */
+export function ReactQueryStreamedHydration(props: {
+  children: React.ReactNode
+  queryClient?: QueryClient
+  transformer?: HydrationStreamProviderProps<DehydratedState>['transformer']
+}) {
+  const queryClient = useQueryClient(props.queryClient)
+
+  // <server only>
```

Comment:
What's this? This code doesn't really only run on the server pass right, it just happens to not do anything in the end when it runs on the client? If the code itself still runs I find this comment a bit confusing.

Reply:
I've moved the comment further down: ba57ad84, but I think the intent was that `trackedKeys`, while it does run on the client as well, is only used on the server:
- the write is in the `typeof window === 'undefined'` check
- the read is in the `onFlush` callback, which also happens on the server only.

Reply:
👍